Core/Src/porting/common.c

```diff
@@ -153,6 +161,11 @@ void common_emu_input_loop(odroid_gamepad_state_t *joystick, odroid_dialog_choic
     static bool macro_activated = false;
     static uint8_t clear_frames = 0;
 
+    void _repaint() {
```

Comment:
not sure if we're cool with function nesting. alternatively we could define this externally as:

```
static void ingame_overlay_loop_then_repaint(void_callback_t repaint){
    ingame_overlay_loop();
    repaint();
}
```

Reply:
I'm also fine with this being nested; it's not like we need to be a coding snob at this point in the project :P

Reply:
I totally understand.
It is a bit controversial.

I am open for alternatives and the reasoning for using them is that it was easy and faster to implement/refactor with very little chance of creating new bugs since they are executed within scope.

This particular one however don't actually need the nesting and could be moved out of scope. I suggest we move it out at a later point not to further push this PR into the future if that is ok.